Public/Src/Cache/ContentStore/Distributed/NuCache/ContentLocationDatabaseExtensions.cs

```diff
@@ -12,6 +15,27 @@ namespace BuildXL.Cache.ContentStore.Distributed.NuCache
     /// </summary>
     public static class ContentLocationDatabaseExtensions
     {
+        /// <summary>
+        /// Performs a compare exchange operation on metadata, while ensuring all invariants are kept. If the
+        /// fingerprint is not present, then it is inserted.
+        /// </summary>
+        /// <returns>
+        /// Result providing the call's completion status. True if the replacement was completed successfully, false otherwise.
+        /// </returns>
+        public static Possible<bool> CompareExchange(

```

Comment:
Do we want to start using 'Possible' types from BuildXL.Utilities? 

I think it is a bit confusing to introduce yet another result type here...

Reply:
It seems that the type was introduced already... But the question still holds...

---
In reply to: [345368562](https://github.com/microsoft/BuildXL/pull/1200#discussion_r345368562) [](ancestors = 345368562)

Reply:
just curious, why not to use BoolResult here instead?

---
In reply to: [345368790](https://github.com/microsoft/BuildXL/pull/1200#discussion_r345368790) [](ancestors = 345368790,345368562)

Reply:
This is code I originally wrote a few months ago. Used CompareExchange because I didn't know how to convert to BoolResult

---
In reply to: [345369015](https://github.com/microsoft/BuildXL/pull/1200#discussion_r345369015) [](ancestors = 345369015,345368790,345368562)